Assets/Scripts/GameManager.cs

```diff
@@ -68,15 +85,47 @@ public class GameManager : MonoBehaviour {
 
     private void FillUITexts() {
         if (!currentTextAlreadyFilled) {
+            Statement CurrQuestion = CurrentQuestion;
+            Answer CurrAnswer;
+            Condition CurrCondition;
+            bool activateButton;
             StopAllCoroutines();
-            StartCoroutine(TypeSentence(CurrentQuestion.Question));
-            for (var i = 0; i < CurrentQuestion.Answers.Length; i++) {
-                ButtonTexts[i].text = CurrentQuestion.Answers[i].Label;
-                AnswerButtons[i].gameObject.SetActive(true);
+            StartCoroutine(TypeSentence(CurrQuestion.Question));
+            for (var i = 0; i < CurrQuestion.Answers.Length; i++) {
+                CurrAnswer = CurrQuestion.Answers[i];
+                CurrCondition = CurrAnswer.Required;
+                activateButton = false;
+                // check for mising key
+                if (CurrCondition.Key != null)
+                {
+                    //check if key is present in dictionary
+                    if (flags.ContainsKey(CurrCondition.Key))
+                    {
+                        //check if flag value matches required value
+                        if (flags[CurrCondition.Key] == CurrCondition.BoolValue)
+                        {
+                            activateButton = true;
+                        }
+                    }
+                    else
+                    {
+                        activateButton = true;
+                    }
+                }
+                else
+                {
+                    activateButton = true;
+                }
+
+                if (activateButton)
+                {
+                    ButtonTexts[i].text = CurrAnswer.Label;
+                    AnswerButtons[i].gameObject.SetActive(true);
+                }
```

Comment:
Since the method "FillUITexts" is getting quite big, this should be refactored into a private function to separate logic and make the code more maintainable